llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -381,6 +383,26 @@ class PropagateIncomingLatencies : public ScheduleDAGMutation {
           }))
         continue;
 
+      // Do not change the latency if the REG_SEQUENCE has one source
+      // outside this MBB. Such REG_SEQUENCE instructions will typically require
+      // a COPY from the external source to one of the lanes of the destination
+      // register. It is important to then keep the REG_SEQUENCE close to its
+      // consumers, because after MachinePipeliner, this typically means that
+      // only the lanes corresponding to internal sources will be loop-carried.
+      // The external lane will come directly from the pre-header, and the
+      // required COPY can then be hoisted by MachineLICM.
+      const MachineBasicBlock &MBB = *MI.getParent();
+      const MachineRegisterInfo &MRI = DAG->MRI;
+      auto HasExternalAndLocalSources = [&MBB, &MRI](const MachineInstr &MI) {
+        return MI.isRegSequence() && MRI.isSSA() && MI.getNumOperands() > 3 &&
+               count_if(MI.uses(), [&MBB, &MRI](const MachineOperand &MO) {
+                 return MO.isReg() && MO.getReg().isVirtual() &&
```

Comment:
check: as we eliminate REG_SEQUENCEs as part of the de-ssa process, do we really need to check `MRI.isSSA()` and `MO.getReg().isVirtual()`?